### PR TITLE
Hermes Python Bindings

### PR DESCRIPTION
**Pull request by anthonyray**

### CODE REVIEW

**Comment by anthonyray** [comment]
Fixed the issues you pointed out @fredszaq 
Can you review ?

**Comment by anthonyray** [comment]
Did some QA, it should work.

File: platforms/hermes-python/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Hermes Python 
+[![Build Status](https://travis-ci.org/snipsco/hermes-protocol.svg?branch=hermes-python-ffi)](https://travis-ci.org/snipsco/hermes-protocol)
```
Comment (fredszaq) [inline platforms/hermes-python/README.md]: need to remove the `?branch=hermes-python-ffi` part

File: platforms/hermes-python/generate_documentation.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                        output.append(pydoc.inspect.getdoc(mth))
+                        output.append("```")
+                    output.append("")
+
+    output_md = "\n".join(output)
+    return output_md
+
+
+if __name__ == "__main__":
+    print generate_docs()
```
Comment (fredszaq) [inline platforms/hermes-python/generate_documentation.py]: missing newline at the end of the file

File: platforms/hermes-python/hermes_python/ffi/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from . import ontology
+from . import utils
```
Comment (fredszaq) [inline platforms/hermes-python/hermes_python/ffi/__init__.py]: missing newline at the end of the file

File: platforms/hermes-python/hermes_python/ffi/utils.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+hermes_drop_hotword_facade = wrap_library_call(lib.hermes_drop_hotword_facade)
+hermes_drop_nlu_backend_facade = wrap_library_call(lib.hermes_drop_nlu_backend_facade)
+hermes_drop_nlu_facade = wrap_library_call(lib.hermes_drop_nlu_facade)
+hermes_drop_sound_feedback_backend_facade = wrap_library_call(lib.hermes_drop_sound_feedback_backend_facade)
+hermes_drop_sound_feedback_facade = wrap_library_call(lib.hermes_drop_sound_feedback_facade)
+hermes_drop_tts_backend_facade = wrap_library_call(lib.hermes_drop_tts_backend_facade)
+hermes_drop_tts_facade = wrap_library_call(lib.hermes_drop_tts_facade)
+
+hermes_protocol_handler_new_mqtt = wrap_library_call(lib.hermes_protocol_handler_new_mqtt)
+hermes_protocol_handler_dialogue_facade = wrap_library_call(lib.hermes_protocol_handler_dialogue_facade)
```
Comment (fredszaq) [inline platforms/hermes-python/hermes_python/ffi/utils.py]: missing newline at the end of the file

File: platforms/hermes-python/hermes_python/hermes.py [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+                                                               CSessionEndedMessage, c_void_p, SessionEndedMessage)
+        hermes_dialogue_subscribe_session_ended(self._facade, self._c_callback_subscribe_session_ended)
+        return self
+
+
+    def subscribe_intent_not_recognized(self, user_callback_subscribe_intent_not_recognized):
+        """
+        Register a callback when the Dialogue Manager doesn't recognize an intent.
+
+        Note that you need to have initialized a session with the intent_not_recognized field set to true.
```
Comment (fredszaq) [inline platforms/hermes-python/hermes_python/hermes.py]: the flag can also be set on the continue session, and always concerns only the next dialogue turn

**Comment by anthonyray** [comment]
Thanks for the review @fredszaq ! Anything else before we merge ?

**Comment by anthonyray** [comment]
Ready to merge ?

**Pull request merged**